dotcom-rendering/src/server/lib/logging-store.ts

```diff
@@ -39,6 +40,32 @@ export const recordTypeAndPlatform = (
 	}
 };
 
+export const recordPageId = (pageId: string): void => {
+	const { request } = loggingStore.getStore() ?? {};
+
+	if (request) {
+		request.pageId = pageId;
+	}
+};
+
+type RecordablePeriod = 'total' | 'json' | 'enhance' | 'render';
+
+export const recordTimeStart = (metric: RecordablePeriod): void => {
+	const { timing } = loggingStore.getStore() ?? {};
+
+	if (timing) {
+		timing[metric] = Date.now();
```

Comment:
Much like in browsers, [`performance.now()` is available](https://nodejs.org/api/perf_hooks.html)!

Reply:
There's some really [nice docs](https://nodejs.org/api/perf_hooks.html#performance-measurement-apis) on the NodeJS performance stuff.

Reply:
I think as @AshCorr  pointed out that the other performance methods would be shared between invocations/requests sent to the same server.